src/main/java/tech/pegasys/tools/epchecks/JavaCase.java

```diff
@@ -0,0 +1,160 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.tools.epchecks;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.bugpatterns.BugChecker.ClassTreeMatcher;
+import static com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import static com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.lang.model.element.Modifier;
+
+import com.google.auto.service.AutoService;
+import com.google.common.base.CaseFormat;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.ModifiersTree;
+import com.sun.source.tree.VariableTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+    name = "JavaCase",
+    summary = "Use the appropriate case.",
+    severity = SUGGESTION,
+    linkType = BugPattern.LinkType.NONE)
+public class JavaCase extends BugChecker
+    implements MethodTreeMatcher, ClassTreeMatcher, VariableTreeMatcher {
+
+  private static final Pattern PATTERN_HAS_LOWER = Pattern.compile("^.*[a-z].*$");
+  private static final Pattern PATTERN_HAS_UPPER = Pattern.compile("^.*[A-Z].*$");
+  private static final Pattern PATTERN_STARTS_WITH_LOWER = Pattern.compile("^[a-z].*$");
+  private static final Pattern PATTERN_STARTS_WITH_UPPER = Pattern.compile("^[A-Z].*$");
+  private static final Pattern PATTERN_UNDERSCORES = Pattern.compile("^_+$");
+
+  @Override
+  public Description matchVariable(VariableTree tree, VisitorState state) {
+    String name = tree.getName().toString();
+    ModifiersTree modifiers = tree.getModifiers();
+    Set<Modifier> flags = modifiers.getFlags();
+
+    // Constants should be all uppercase.
+    if (flags.contains(Modifier.STATIC) && flags.contains(Modifier.FINAL)) {
+      if (!isUpperUnderscore(name)) {
+        return buildDescription(tree)
+            .addFix(SuggestedFixes.renameVariable(tree, toUpperUnderscore(name), state))
+            .build();
+      }
+    } else if (!isUnderscore(name) && !isLowerCamel(name)) {
+      return buildDescription(tree)
+          .addFix(SuggestedFixes.renameVariable(tree, toLowerCamel(name), state))
+          .build();
+    }
+    return Description.NO_MATCH;
+  }
+
+  @Override
+  public Description matchMethod(MethodTree tree, VisitorState state) {
+    String name = tree.getName().toString();
+
+    // Class constructors have the <init> name.
+    if (name.equals("<init>")) return Description.NO_MATCH;
```

Comment:
nit: We generally use braces on all if statements:
```suggestion
    if (name.equals("<init>")) {
      return Description.NO_MATCH;
    }
```

Reply:
Will update. I totally understand. I like to keep things consistent too.